File: battery.py
#!/usr/bin/env python3
import sys
import psutil
from time import sleep

def polyPrint(string):
    print(string, flush=True)  # Flush to ensure Polybar reads it immediately

def showBattery():
    battery = psutil.sensors_battery()
    plugged = battery.power_plugged
    percent = int(battery.percent)
    if plugged and percent == 100:
        return f"{percent}%"
    elif plugged:
        return f"{percent}% Charging"
    return f"{percent}%"

while True:
    polyPrint(showBattery())
    sleep(2)

File: battery.lua
local polyprint = require("polyprint")
local function getBatteryInfo()
    -- Read battery percentage
    local capacity = io.open("/sys/class/power_supply/BAT1/capacity", "r")
    if not capacity then return end
    local percent = capacity:read("*n")  -- Read number
    capacity:close()

    -- Read charging status
    local status = io.open("/sys/class/power_supply/BAT1/status", "r")
    if not status then return end
    local charging = status:read("*l")
    status:close()

    if charging == "Full" and percent == 100 then
        return percent .. "%"
    elseif charging == "Charging" then
        return percent .. "% Charging"
    else
        return percent .. "%"
    end
end

while true do
    polyprint(getBatteryInfo())
    os.execute("sleep 1")
end
